eureka-client/src/main/java/com/netflix/eureka2/client/registration/RegistrationHandlerImpl.java

```diff
@@ -59,7 +59,7 @@ public RegistrationHandlerImpl(ClientChannelFactory channelFactory) {
             logger.info("Instance: {} is not registered. Ignoring unregister", instanceInfo);
             return Observable.empty(); // Be more acceptable to errors from user as unregister for non-existent instance is a no-op.
         }
-        return registrationChannel.unregister().doOnTerminate(new Action0() {
```

Comment:
It's preferable that we do retries for client side unregister failures instead of just walking away from the connection. Otherwise, the instance will be unregistered via the eviction mechanism which can be slow.